src/Nethermind/Nethermind.State/OverridableWorldStateManager.cs

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Db;
+using Nethermind.Logging;
+using Nethermind.Trie.Pruning;
+
+namespace Nethermind.State;
+
+public class OverridableWorldStateManager : IWorldStateManager
+{
+    private readonly IDb _codeDb;
+    private readonly StateReader _reader;
+    private readonly WorldState _state;
+
+    private readonly OverlayTrieStore _overlayTrieStore;
+    private readonly ILogManager? _logManager;
+
+    public OverridableWorldStateManager(IDbProvider dbProvider, IReadOnlyTrieStore trieStore, ILogManager? logManager)
+    {
+        dbProvider = new ReadOnlyDbProvider(dbProvider, true);
+        OverlayTrieStore overlayTrieStore = new(dbProvider.StateDb, trieStore, logManager);
+
+        _logManager = logManager;
+        _codeDb = dbProvider.GetDb<IDb>(DbNames.Code);
+        _reader = new(overlayTrieStore, dbProvider.GetDb<IDb>(DbNames.Code), logManager);
+        _state = new(overlayTrieStore, dbProvider.GetDb<IDb>(DbNames.Code), logManager);
+        _overlayTrieStore = overlayTrieStore;
+    }
+
+    public IWorldState GlobalWorldState => _state;
+    public IStateReader GlobalStateReader => _reader;
+    public IReadOnlyTrieStore TrieStore => _overlayTrieStore.AsReadOnly();
+
+    public IWorldState CreateResettableWorldState(IWorldState? forWarmup = null)
+    {
+        // TODO add if needed?
+        if (forWarmup is not null)
```

Comment:
As of now `OverridableWorldStateManager.CreateResettableWorldState()` with non-null `forWarmup` is not used, but this `NotSupportedException` also doesn't look good.

Implementing missing part is a bit problematic due to `PreCachedTrieStore` not having an alternative for `TrieStore.ClearCache()`, which is used by `OverridableWorldState`.

Reply:
`PreCachedTrieStore` need to be wrapped into `OverlayTrieStore` to do a post-call cleanup, but latter requires `IReadOnlyTrieStore` interface.

It probably can be fixed in some way, just not sure it's worth the trouble at the moment.